src/api/assets.go

```diff
@@ -3,26 +3,67 @@ package api
 import (
 	"fmt"
 	"html/template"
+	"net"
 	"net/http"
 
 	"github.com/gorilla/mux"
 )
 
-func rootHandler(w http.ResponseWriter, r *http.Request) {
-	data, err := Asset("static/index.htm")
-	if err != nil {
-		fmt.Println(err)
-		w.WriteHeader(404)
-		return
+type bindInfo struct {
+	ApiUrl string
+	IP4    string
+	IP6    string
+	Port   string
+}
+
+type templateData struct {
+	ApiUrl    string
+	Primary   string
+	Secondary string
+}
+
+func (info *bindInfo) templateFor(r *http.Request) *templateData {
+	data := &templateData{
+		ApiUrl: info.ApiUrl,
 	}
-	tmpl, err := template.New("root").Parse(string(data))
-	if err != nil {
-		fmt.Println(err)
-		w.WriteHeader(500)
-		return
+	addr, _, _ := net.SplitHostPort(r.RemoteAddr)
+	if len(net.ParseIP(addr)) == net.IPv6len {
+		// client is ipv6
+		data.Primary = info.IP6
+		data.Secondary = info.IP4
+	} else {
+		// client is ipv4
+		data.Primary = info.IP4
+		data.Secondary = info.IP6
+	}
+	// If the user didn't provide one of the two addresses, we pass the UI an
+	// empty string.
+	if data.Primary != "" {
+		data.Primary = net.JoinHostPort(data.Primary, info.Port)
+	}
+	if data.Secondary != "" {
+		data.Secondary = net.JoinHostPort(data.Secondary, info.Port)
+	}
```

Comment:
Nothing here is hardcoding us to HTTP vs HTTPS. And providing the ports here doesn't cause any issues. It just means the UI is going to be more specific than it needs to be with the AJAX calls. Given that, I don't think it buys us very much to put extra logic in here to prevent adding the port numbers if they're unnecessary. Having port numbers works in 100% of cases, so I don't see the isse.